docs/proposals/push.md

```diff
@@ -0,0 +1,152 @@
+<!-- BEGIN MUNGE: UNVERSIONED_WARNING -->
+
+<!-- BEGIN STRIP_FOR_RELEASE -->
+
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+<img src="http://kubernetes.io/kubernetes/img/warning.png" alt="WARNING"
+     width="25" height="25">
+
+<h2>PLEASE NOTE: This document applies to the HEAD of the source tree</h2>
+
+If you are using a released version of Kubernetes, you should
+refer to the docs that go with that version.
+
+<!-- TAG RELEASE_LINK, added by the munger automatically -->
+<strong>
+The latest release of this document can be found
+[here](http://releases.k8s.io/release-1.3/docs/proposals/push.md).
+
+Documentation for other releases can be found at
+[releases.k8s.io](http://releases.k8s.io).
+</strong>
+--
+
+<!-- END STRIP_FOR_RELEASE -->
+
+<!-- END MUNGE: UNVERSIONED_WARNING -->
+
+# Proposal: commit a container and push it to registry
+----
+
+## Motivation
+
+There are some use case to commit a container, and push its resulting image to
+a registry, for example:
+
+1. For a run-once pod whose result represents an image that should be pushed to
+a registry for consumption by other actors in the system
+
+2. There is someone desire to be able to modify a running container and then
+commit/push the change so they could, for example, make a debug change and then
+scale up the deployment to test it in a clustered mode, without having to start
+from scratch with building a new image.
+
+The goal of this proposal is to add the commit/push capability, which will allow
+user commit and push their containers manually, or automically before/after
+container exits.
+
+## Design
+
+### Manually commit and push a container
+
+######1. create a ThirdPartyResource
+
+```
+metadata:
+  name: push.k8s.io
+apiVersion: extensions/v1beta1
+kind: ThirdPartyResource
+description: "Allow user to commit a container and push it into registry"
+versions:
+- name: v1
+```
+
+###### 2. create a `Push` resource
```

Comment:
The name Push feels more like a verb, is there any other options that you've considered?